### DISCUSSION TITLE
dynamic=auto doesn't generate static files/use Next.js cache

### DISCUSSION THREAD

**Discussion by pmi** [vercel/next.js#63979] [Help]
### Summary

Next.js 14.1.0

I have a page with `generateStaticParams` that produces html+meta+rsc files for every content item in `.next/server/app/...` folder at build time when `dynamic=force-static` or `dynamic=error`. 

But whenever I set `dynamic=auto` it doesn't generate static files and I can see in logs that each request is processed at runtime:

- dynamic=auto
  - Build logs say pages are SSG  
  - No html + meta + rsc files created in `.next/server/app/...` folder
  - Next.js cache header is absent 
- dynamic=force-static
  - Build logs say pages are SSG  
  - Files created in `.next/server/app/...` folder
  - Next.js cache HIT header is present

I don't use any dynamic functions that could bail out static generation.
I use fetch with POST request to graphql endpoint without any custom cache settings.
I use middleware with rewrite invocation if that makes any difference.

Looking at the docs I would assume that Next.js would cache the data + views and create static pages in `dynamic=auto` mode as well ?
Also, is there is a way to tell if the data fetch results were cached ?

Thanks in advance !

### Additional information
**Page.tsx**
```
export type PageProps = {
    locale: string,
    contentPath: string[],
}

export default async function Page({params}: { params: PageProps }) {

    console.info(`Accessing page`, params);

    const start = Date.now();
    const data: FetchContentResult = await fetchContent(params);
    const duration = Date.now() - start;

    console.info(`Page fetch took ${duration} ms`);

    validateData(data);

    return (
        <MainView {...data}/>
    )
};

export async function generateMetadata({params}: { params: PageProps }): Promise<Metadata> {
    const {common} = await fetchContent(params);
    return {
        title: common?.get?.displayName || 'Not found',
    };
}

export async function generateStaticParams(props: { params: PageProps }): Promise<any[]> {
    return await fetchContentPathsForAllLocales('\${site}/');
}
```

where `fetchContent` is a complex function posting to graphql endpoint with `fetch`:
```
    const opts = {
        method: options?.method || 'POST',
        body: options?.body ? JSON.stringify(options?.body) : null,
        headers: mergeHeaders(projectConfig, options?.headers),
        next: options?.next,
        cache: options?.cache,
    };

    try {
        const res = await fetch(apiUrl, opts);
    }
```

and `validateData` basically throws `notFound` if there is no content.

**Comment by Rishab49** [comment]
can you share the page code?

**Comment by pmi** [comment]
Hi @Rishab49 !

I have now added page source to the description !

**Comment by kurtrank** [comment]
I ran into this myself recently, and I think there is a bug and/or the docs are incorrect about `auto`. I found an existing issue about it here https://github.com/vercel/next.js/issues/65799

> Also, is there is a way to tell if the data fetch results were cached ?

With 16.3.0 it looks like they are stored in `.next/cache/fetch-cache`, though maybe that has changed in the last 2 years since you posted this haha.
